src/library_pthread.js

```diff
@@ -42,7 +42,6 @@ var LibraryPThread = {
 
       // Allocate memory for thread-local storage.
       var tlsMemory = {{{ makeStaticAlloc(cDefine('PTHREAD_KEYS_MAX') * 4) }}};
-      for (var i = 0; i < {{{ cDefine('PTHREAD_KEYS_MAX') }}}; ++i) HEAPU32[tlsMemory/4+i] = 0;
```

Comment:
This change worries me. Is the assumption the memory was calloc'd elsewhere? We can destroy and reuse threads, though, so I think we have to re-initialize this memory here?

Reply:
My assumption was that since this memory is allocated at compile time, and since ArrayBuffers are initialised as 0, then setting this to 0 was redundant.
I didn't think about this being re-initialised, but I agree, that could (and probably will) be a problem.
Perhaps why those tests failed?